### PR TITLE
Error responses through error variables

### PR DESCRIPTION
**Pull request by dutterbutter**
This is a draft pull request to review the possibility of having an `errors.go` file within each package to provide improved meaning to errors. It could also make testing error handeling within functions easier.

This approach can be utilized in different ways, where the variable error is returned in a function, and then used as a part of a switch case or if condition. For instance, 

```
	it := newInsertIterator(...args)
	block, err := it.next()

	if err == ErrKnownBlock {
		// do something with known error
	}
```

## Changes
- Added `errors.go` in `polkadb/`
- 
- 

## Tests:
```

```

### Fixes:

### CODE REVIEW

File: codec/encode_test.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ var encodeTests = []encodeTest{
 	{val: []int{1073741824, 2, 3, 4}, output: []byte{0x10, 0x03, 0x00, 0x00, 0x00, 0x40, 0x08, 0x0c, 0x10}, bytesEncoded: 9},
 	{val: []int{1 << 32, 2, 3, 1 << 32}, output: []byte{0x10, 0x07, 0x00, 0x00, 0x00, 0x00, 0x01, 0x08, 0x0c, 0x07, 0x00, 0x00, 0x00, 0x00, 0x01}, bytesEncoded: 15},
 	{val: []bool{true, false, true}, output: []byte{0x0c, 0x01, 0x00, 0x01}, bytesEncoded: 4},
-	{val: [][]int{[]int{0, 1}, []int{1, 0}}, output: []byte{0x08, 0x08, 0x00, 0x04, 0x08, 0x04, 0x00}, bytesEncoded: 7},
```
Comment (ansermino) [inline codec/encode_test.go]: Not sure why this is in here...

**Comment by ansermino** [comment]
Is it possible to include values in these? eg. `failed to retrieve key: 0x1234`

File: codec/encode_test.go (hunk 1 above)
Comment (dutterbutter) [reply codec/encode_test.go]: Sorry about that, that shouldn't be there.

**Comment by dutterbutter** [comment]
> Is it possible to include values in these? eg. `failed to retrieve key: 0x1234`

Not unless we make them functions, cause you would need to pass in the value somehow.

**Pull request closed without merging**